src/sbd-inquisitor.c

```diff
@@ -1307,37 +1303,51 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
-        } else if (strcmp(argv[optind], "watch") == 0) {
+        } else
+#endif
+        if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
+        } else {
+            exit_status = -2;
+        }
+
+        if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+                const char *delay_source = "SBD_DELAY_START";
```

Comment:
not necessarily source of delay at that point still can be SBD_DELAY_START, SBD_WATCHDOG_TIMEOUT * 2, default watchdog-timeout * 2, watchdog-timeout from cmdline

Reply:
I'm a bit confused about that.
I see delay set only here after it has been set initially to 0.

value = get_env_option("SBD_DELAY_START");
        if(value) {
            delay_start = crm_is_true(value);

            if (!delay_start) {
                delay = crm_get_msec(value) / 1000;
                if (delay > 0) {
                    delay_start = true;
                }
            }
        }

After that point it has been set in the watch section.

Reply:
True. But if it is not set there the source is 'timeout_watchdog * 2' and that again can have different sources.
Tracking where certain settings are coming from - and logging e.g. in debug or even at higher level together with the value - is one of my projects.
Here debug had that implemented already half-way so I thought we should keep it.
So as to keep that here low effort we can go with 'watchdog-timeout * 2' for now in cases startup-delay is derived from watchdog timeout. We still can add tracking of where watchdog-timeout is coming from in a separate PR including the decision if with delay we still want to show with 'watchdog-timeout' or if we replace the string with the details.

Reply:
Thanks for the clarifications and for your patience to review my changes.
I will commit the relevant changes.